src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -107,6 +123,36 @@ def get_conversion_options(self):
         conversion_options = dict(write_as="raw", es_key="ElectricalSeries_raw", stub_test=False)
         return conversion_options
 
+    def get_device_metadata(self) -> dict:
+        """Returns a device with description including the metadat as described here
+        # https://billkarsh.github.io/SpikeGLX/Sgl_help/Metadata_30.html
+
+        Returns
+        -------
+        dict
+            a dict containing the metadata necessary for creating the device
+        """
+
+        meta = self.meta
+
+        probe_type = str(meta.get("imDatPrb_type", "no probe type"))
+        probe_type_to_probe_description = {"0": "NP1.0", "21": "NP2.0(1-shank)", "24": "NP2.0(4-shank)"}
+        probe_type_description = probe_type_to_probe_description.get(probe_type, "no probe description")
+
```

Comment:
If these fields are missing I'd rather just not populate them over filling them with filler data like this

Reply:
Yes, in the latest version they are not added to the dict structure unless they are not available. 

I also added a test for the device metadata.